sklearn/tests/test_kernel_ridge.py

```diff
@@ -1,85 +1,143 @@
+    assert_array_equal(krc_cloned.predict(iris.data),
+                       krc_builtin.predict(iris.data))
+    assert_array_almost_equal(krc_cloned.predict(iris.data),
+                              krc_builtin.predict(iris.data))
+
+
+def test_krc_bad_kernel():
+    clf = KernelRidgeClassifier(kernel=lambda x, y: x)
+    assert_raises(ValueError, clf.fit, X, Y)
+
+
+def test_bad_input():
+    # Test that it gives proper exception on negative alpha
+    # alpha cannot be negative because Kernel Ridge come from
+    # a minimization problem where all the terms must be positive
+    assert_raises(ValueError, KernelRidgeClassifier(alpha=-0.5).fit, X, Y)
+
+
+# Test for regression
 def test_kernel_ridge():
-    pred = Ridge(alpha=1, fit_intercept=False).fit(X, y).predict(X)
-    pred2 = KernelRidge(kernel="linear", alpha=1).fit(X, y).predict(X)
+    pred = Ridge(alpha=1, fit_intercept=False).fit(X_reg, y_reg).predict(X_reg)
+    pred2 = KernelRidge(kernel="linear",
+                        alpha=1).fit(X_reg, y_reg).predict(X_reg)
     assert_array_almost_equal(pred, pred2)
 
 
 def test_kernel_ridge_csr():
     pred = Ridge(alpha=1, fit_intercept=False,
-                 solver="cholesky").fit(Xcsr, y).predict(Xcsr)
-    pred2 = KernelRidge(kernel="linear", alpha=1).fit(Xcsr, y).predict(Xcsr)
+                 solver="cholesky").fit(X_reg_csr, y_reg).predict(X_reg_csr)
+    pred2 = KernelRidge(kernel="linear",
+                        alpha=1).fit(X_reg_csr, y_reg).predict(X_reg_csr)
     assert_array_almost_equal(pred, pred2)
 
 
 def test_kernel_ridge_csc():
     pred = Ridge(alpha=1, fit_intercept=False,
-                 solver="cholesky").fit(Xcsc, y).predict(Xcsc)
-    pred2 = KernelRidge(kernel="linear", alpha=1).fit(Xcsc, y).predict(Xcsc)
+                 solver="cholesky").fit(X_reg_csr, y_reg).predict(X_reg_csr)
+    pred2 = KernelRidge(kernel="linear",
+                        alpha=1).fit(X_reg_csr, y_reg).predict(X_reg_csr)
     assert_array_almost_equal(pred, pred2)
 
 
 def test_kernel_ridge_singular_kernel():
     # alpha=0 causes a LinAlgError in computing the dual coefficients,
     # which causes a fallback to a lstsq solver. This is tested here.
-    pred = Ridge(alpha=0, fit_intercept=False).fit(X, y).predict(X)
+    pred = Ridge(alpha=0, fit_intercept=False).fit(X_reg, y_reg).predict(X_reg)
     kr = KernelRidge(kernel="linear", alpha=0)
-    ignore_warnings(kr.fit)(X, y)
-    pred2 = kr.predict(X)
+    ignore_warnings(kr.fit)(X_reg, y_reg)
+    pred2 = kr.predict(X_reg)
     assert_array_almost_equal(pred, pred2)
 
 
-def test_kernel_ridge_precomputed():
-    for kernel in ["linear", "rbf", "poly", "cosine"]:
-        K = pairwise_kernels(X, X, metric=kernel)
-        pred = KernelRidge(kernel=kernel).fit(X, y).predict(X)
-        pred2 = KernelRidge(kernel="precomputed").fit(K, y).predict(K)
-        assert_array_almost_equal(pred, pred2)
-
-
-def test_kernel_ridge_precomputed_kernel_unchanged():
```

Comment:
why was this removed?

Reply:
It is not removed; in the same file (_test_kernel_ridge.py_), it appears some tests for classification (_lines from 43 to 86_)  and for regression (_lines from 89 to 143_).